aries_cloudagent/protocols/present_proof/presentation_exchange/pres_exch_handler.py

```diff
@@ -0,0 +1,823 @@
+        count = total_count
+    requirement = Requirement(
+        count=count,
+        maximum=sr.maximum,
+        minimum=sr.minimum,
+        input_descriptors=input_descriptors,
+        nested_req=nested,
+    )
+    return requirement
+
+
+async def make_requirement(
+    srs: Sequence[SubmissionRequirements] = None,
+    descriptors: Sequence[InputDescriptors] = None,
+) -> Requirement:
+    """
+    Return Requirement.
+
+    Creates and return Requirement with nesting if required
+    using to_requirement()
+
+    Args:
+        srs: list of submission_requirements
+        descriptors: list of input_descriptors
+    Raises:
+        PresentationExchError: If not able to create requirement
+
+    """
+    if not srs:
+        srs = []
+    if not descriptors:
+        descriptors = []
+    if len(srs) == 0:
+        requirement = Requirement(
+            count=len(descriptors),
+            input_descriptors=descriptors,
+        )
+        return requirement
+    requirement = Requirement(
+        count=len(srs),
+        nested_req=[],
+    )
+    for tmp_sub_req in srs:
+        try:
+            requirement.nested_req.append(
+                await to_requirement(tmp_sub_req, descriptors)
+            )
+        except Exception as err:
+            raise PresentationExchError(
+                "Error creating requirement " f"inside to_requirement function, {err}"
+            )
+    return requirement
+
+
+async def is_len_applicable(req: Requirement, val: int) -> bool:
+    """
+    Check and validate requirement minimum, maximum and count.
+
+    Args:
+        req: Requirement
+        val: int value to check
+    Return:
+        bool
+
+    """
+    if req.count:
+        if req.count > 0 and val != req.count:
+            return False
+    if req.minimum:
+        if req.minimum > 0 and req.minimum > val:
+            return False
+    if req.maximum:
+        if req.maximum > 0 and req.maximum < val:
+            return False
+    return True
+
+
+async def contains(data: Sequence[str], e: str) -> bool:
+    """
+    Check for e in data.
+
+    Returns True if e exists in data else return False
+
+    Args:
+        data: Sequence of str
+        e: str value to check
+    Return:
+        bool
+
+    """
+    data_list = list(data) if data else []
+    for tmp_item in data_list:
+        if e == tmp_item:
+            return True
+    return False
+
+
+async def filter_constraints(
+    constraints: Constraints, credentials: Sequence[VCRecord]
+) -> Sequence[VCRecord]:
+    """
+    Return list of applicable VCRecords after applying filtering.
+
+    Args:
+        constraints: Constraints
+        credentials: Sequence of credentials
+            to apply filtering on
+    Return:
+        Sequence of applicable VCRecords
+
+    """
+    result = []
+    for tmp_cred in credentials:
+        if (
+            constraints.subject_issuer is not None
+            and constraints.subject_issuer == "required"
```

Comment:
The `is not None` is redundant I guess?

```suggestion
            constraints.subject_issuer == "required"
```